extension/js/browser-polyfill.js

```diff
@@ -0,0 +1,1224 @@
+(function (global, factory) {
```

Comment:
Where did this file come from? It looks generated.
Wondering if we can keep it in sync from its upstream source (if this is in npm or somewhere more definitive).

Reply:
Yeah, it's from [the npm `webextension-polyfill` module](https://www.npmjs.com/package/webextension-polyfill). AFAIK, npm `postinstall` scripts have to be included with the module itself? Is there a way to have our own custom post-install script run that will copy the `browser-polyfill.js` file from `node_modules/` into the `src/` dir?

Reply:
Nope, we can add our own "postinstall" npm script in the extension/package.json file that would automatically copy over the ./node_module/'ed file into ./extension/js/browser-polyfill.js (and then we can add that file to .gitignore).

Alternatively, we can add some "prebuild" hook or tweak how the extension does <kbd>npm run build</kbd> so the file gets copied over from ./node_modules/ at build time.

Reply:
In fact, here's what that might look like:

```js
  "dependencies": {
    "copy": "^0.3.2",
    "web-ext": "^5.3.0",
    "webextension-polyfill": "^0.6.0"
  },
  "scripts": {
    "build:clean": "...",
    "postinstall": "copy './node_modules/webextension-polyfill/dist/browser-polyfill.min.*' ./js/",
    "test": "echo \"Error: no test specified\" && exit 1"
  },
```

I use the npm **copy** module to copy the 2 minified polyfill assets (JavaScript and linked sourcemap) from ./node_modules/webextension-polyfill/dist/ to the ./extension/js/ folder after you run <kbd>npm install</kbd>.
We could also add it to the `build:clean` script, but that might not trigger if you run <kbd>npm run build</kbd> directly, and sometimes <kbd>npm run build:clean</kbd> isn't run until after the add-on is built, so the timing can be a bit tricky.

Reply:
Also looks like we can shave about 26-30KB of bandwidth by using the .min.js versions instead of the non-minified counterparts:

```sh
ls -lh ./node_modules/webextension-polyfill/dist 

-rw-r--r--  1 pdehaan  staff    36K 26 Oct  1985 browser-polyfill.js
-rw-r--r--  1 pdehaan  staff    56K 26 Oct  1985 browser-polyfill.js.map
-rw-r--r--  1 pdehaan  staff   9.9K 26 Oct  1985 browser-polyfill.min.js
-rw-r--r--  1 pdehaan  staff    53K 26 Oct  1985 browser-polyfill.min.js.map
```